internal/controller/pullrequest_controller.go

```diff
@@ -58,121 +58,92 @@ type PullRequestReconciler struct {
+	logger.Info("Reconciling PullRequest state", "desired", pr.Spec.State, "current", pr.Status.State)
+	switch pr.Spec.State {
+	case promoterv1alpha1.PullRequestOpen:
+		if pr.Status.State != promoterv1alpha1.PullRequestOpen {
+			logger.Info("Creating PullRequest")
+			if err := r.createPullRequest(ctx, &pr, provider); err != nil {
+				return ctrl.Result{}, fmt.Errorf("failed to create pull request: %w", err)
+			}
 		}
-		// We requeue here because we want to delete the resource on the next reconcile, this does cause more API calls to the provider
-		// but is cleaner than deleting right away. Will revisit this.
-		return ctrl.Result{Requeue: true}, nil
-	}
-
-	// We want to close the PR, but it's not closed on the provider based on status, we should close it on provider and delete it from k8s.
-	if pr.Spec.State == promoterv1alpha1.PullRequestClosed && pr.Status.State != promoterv1alpha1.PullRequestClosed {
-		err := r.closePullRequest(ctx, &pr, pullRequestProvider)
-		if err != nil {
-			return ctrl.Result{}, fmt.Errorf("failed to close pull request: %w", err)
+	case promoterv1alpha1.PullRequestMerged:
+		if pr.Status.State != promoterv1alpha1.PullRequestMerged {
+			logger.Info("Merging PullRequest")
+			if err := r.mergePullRequest(ctx, &pr, provider); err != nil {
+				return ctrl.Result{}, fmt.Errorf("failed to merge pull request: %w", err)
+			}
+			// We do not need to update status here because we close the PR on the SCM but we requeue the reconcile to
+			// clean up the PullRequest object
+			return ctrl.Result{Requeue: true}, nil
 		}
-
-		err = r.Status().Update(ctx, &pr)
-		if err != nil {
-			return ctrl.Result{}, fmt.Errorf("closed pull request, but failed to update PullRequest status: %w", err)
+	case promoterv1alpha1.PullRequestClosed:
+		if pr.Status.State != promoterv1alpha1.PullRequestClosed {
+			logger.Info("Closing PullRequest")
+			if err := r.closePullRequest(ctx, &pr, provider); err != nil {
+				return ctrl.Result{}, fmt.Errorf("failed to close pull request: %w", err)
+			}
+			// We do not need to update status here because we close the PR on the SCM but we requeue the reconcile to
+			// clean up the PullRequest object
+			return ctrl.Result{Requeue: true}, nil
 		}
-		// We requeue here because we want to delete the resource on the next reconcile, this does cause more API calls to the provider
-		// but is cleaner than deleting right away. Will revisit this.
-		return ctrl.Result{Requeue: true}, nil
 	}
 
 	if pr.Status.ObservedGeneration != pr.Generation {
-		// TODO: is there more work avoidance we can do here? Just because the generation has changed, do we know that
-		//       we need to hit the SCM? For example, an annotation update probably shouldn't cause an API call.
-		err = r.updatePullRequest(ctx, pr, pullRequestProvider)
-		if err != nil {
+		if err := r.updatePullRequest(ctx, pr, provider); err != nil {
```

Comment:
I think we hit this line on an initial PR opening, causing an unnecessary SCM call. Maybe also check `pr.Status.ObservedGeneration != ""` to detect a newly-created CR? I don't like the obscurity of that check, but would solve the problem.